data_pipeline/surveymonkey/surveymonkey_etl.py

```diff
@@ -38,34 +38,31 @@ def get_survey_question_details(access_token: str, survey_id: str) -> list:
 
 
 def get_no_answer_options_for_question_json() -> dict:
+    """Added for consistency of the schema in BQ"""
     return {
-        "id": None,
-        "text": None,
-        "type": None,
-        "weight": None
+        "id": "",
+        "text": "",
+        "type": "",
+        "weight": ""
     }
 
 
 def parse_answer_options_in_question_answer_json(
-    question_answer_response_json: dict,
-    option_key: str
+    question_answer_response_json: dict
 ) -> dict:
-    if isinstance(question_answer_response_json[option_key], list):
-        result_list = []
-        for answer_option in question_answer_response_json[option_key]:
-            result_list.append({
-                "id": answer_option.get("id"),
-                "text": answer_option.get("text"),
-                "type": answer_option.get("type"),
-                "weight": answer_option.get("weight")
-            })
-        return result_list
-    return [{
-        "id": question_answer_response_json[option_key].get("id"),
-        "text": question_answer_response_json[option_key].get("text"),
-        "type": question_answer_response_json[option_key].get("type"),
-        "weight": question_answer_response_json[option_key].get("weight")
-    }]
+    if isinstance(question_answer_response_json, list):
+        answer_options = question_answer_response_json
+    else:
+        answer_options = [question_answer_response_json]
+    return [
+        {
+            "id": str(answer_option.get("id", "")),
+            "text": str(answer_option.get("text", "")),
+            "type": str(answer_option.get("type", "")),
+            "weight": str(answer_option.get("weight", ""))
```

Comment:
In addition to the comment regarding converting weight to a str. In general while converting to str works, it would mask errors. It would be good to have actual examples where the types are inconsistent. e.g. in what case is `weight` an int or another type? Same for the other fields.
It would be better if it fails so we can look at it.

Reply:
OK

Reply:
`None` is causing problem (while creating schema and loading the data in BQ) if I don't add a default value in the `get` function? What can I add here as default value, for string values I was adding `""`. Or do you have any idea how can I manage the None value in json not to cause a problem.

```python
"id": answer_option.get("id",""),
"text": answer_option.get("text",""),
"type": answer_option.get("type",""),
"weight": answer_option.get("weight")
```

Reply:
I see your response now here. In general BigQuery fields can be optional so that shouldn't be an issue in general.
I think it would be useful to see what schema is being generated that is causing the issue, and what the exact error is.
It could be that the schema generation or JSON output doesn't work well with `None` fields. And it may work better if the field isn't there at all.

Reply:
I gave it a quick test with some dummy data:

```python
import json
from io import StringIO
from bigquery_schema_generator.generate_schema import SchemaGenerator

generator = SchemaGenerator(
    input_format="json",
    quoted_values_are_strings=True
)
json_data = {"data": [{"text": "value1", "weight": 1}, {"text": "value1", "weight": None}]}
json_str = json.dumps(json_data)
print('json: %r' % json_str)
schema_map, _ = generator.deduce_schema(StringIO(json_str))
schema = generator.flatten_schema(schema_map)
print('schema: %s' % schema)
```

Output:

```text
json: '{"data": [{"text": "value1", "weight": 1}, {"text": "value1", "weight": null}]}'
schema: [OrderedDict([('fields', [OrderedDict([('mode', 'NULLABLE'), ('name', 'text'), ('type', 'STRING')]), OrderedDict([('mode', 'NULLABLE'), ('name', 'weight'), ('type', 'INTEGER')])]), ('mode', 'REPEATED'), ('name', 'data'), ('type', 'RECORD')])]
```

This shows that in this case `None` is serialized as `null`.
It doesn't seem to trip over the schema generator. It is actually declaring both `text` and `weight` as `NULLABLE`. Which is probably good as it allows us to deprecate fields over time.

Reply:
Just adding a comment that it seems that it was the BigQuery import that was failing with `null` values.
We changed the code to not include None fields. i.e. the answer option dictionary will only have non-none values.
As a result, the resulting BigQuery schema may only show a subset of the four fields, depending on the question type / parent answer field (e.g. `choices` contains `weight`, but others might not)